src/storage-types/src/controller.rs

```diff
@@ -331,4 +336,17 @@ impl TxnsCodec for TxnsCodecRow {
             TxnsEntry::Append(data_id, batch.unwrap_bytes().to_vec())
         }
     }
+
+    fn should_fetch_part(data_id: &ShardId, stats: &PartStats) -> Option<bool> {
+        fn col<'a, T: Data>(stats: &'a StructStats, col: &str) -> Option<&'a T::Stats> {
+            stats
+                .col::<T>(col)
+                .map_err(|err| error!("unexpected stats type for col {}: {}", col, err))
+                .ok()?
+        }
+        let stats = col::<Option<DynStruct>>(&stats.key, "ok")?;
```

Comment:
Is the transactions shard serialized as `SourceData`? (Sounds good if so; just wasn't able to find the answer quickly.)

Reply:
Yup! (It's the bit just above this)